### PR TITLE
Add agent classloader

### LINKED ISSUE #9: A bug along the loading path of PluginClassloader
For loading classes contained in JavaAgent Jar,  `PluginClassLoader` now attempts to delegate to `SystemClassLoader` to resolve them.  It leads some classes other than what in JavaAgent Jar to be    
resolved by `PluginClassLoader`. It should be safer to only resolve among classpath of JavaAgent.

### refer
+ [How to get vm arguments from inside of java application?
](https://docs.oracle.com/javase/7/docs/api/java/lang/management/RuntimeMXBean.html#getInputArguments())

+ [-javaagent:jarpath[=options]](https://docs.oracle.com/javase/6/docs/api/java/lang/instrument/package-summary.html)

### PR DESCRIPTION
**Pull request by QilongZhang**
This commit fix  #9

### CODE REVIEW

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/service/classloader/ClassloaderServiceImpl.java [hunk 1]
```diff
@@ -44,10 +48,13 @@
 public class ClassloaderServiceImpl implements ClassloaderService {
 
     private static final ArkLogger LOGGER = ArkLoggerFactory.getDefaultLogger();
+    private static final String JAVA_AGENT_MARK = "-javaagent:";
+    private static final String JAVA_AGENT_OPTION_MARK = "=";
```
Comment (straybirdzls) [inline sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/service/classloader/ClassloaderServiceImpl.java]: Seems the code formatter plugin stop working... codes are not formatted as they should be.

Comment (QilongZhang) [reply sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/service/classloader/ClassloaderServiceImpl.java]: That's too funny, Only this file can NOT formatted correctly.

**Comment by khotyn** [comment]
@QilongZhang You should set the your email and name of this repository to the github email and username, otherwise, your contribution to this project will not be calculated by Github.

image

**Comment by khotyn** [comment]
@QilongZhang @straybirdzls Should we add a check of the format of the files in travis as we did in gitlab? so the unformatted code can never be merged into master.

**Comment by QilongZhang** [comment]
@khotyn   now an enforced check of code format would be executed before being merged to master.

**Pull request merged**